### PR TITLE
Sail default for quickstarts

### PR DESCRIPTION
**Pull request by jasonmadigan**
- Default quickstart-setup.sh to Istio installs via Sail
- Also needs a docs update to remove ref to the env-var.

### CODE REVIEW

File: hack/quickstart-setup.sh [hunk 1]
```diff
@@ -450,21 +446,10 @@ success "Gateway API installed successfully."
...
 info "Installing Istio as a Gateway API provider... 🛫"
-if [ "$ISTIO_INSTALL_SAIL" = true ]; then
-  info "Installing Istio via Sail"
-  kubectl apply -k ${KUADRANT_ISTIO_KUSTOMIZATION}
-  kubectl -n istio-system wait --for=condition=Available deployment istio-operator --timeout=300s
-  kubectl apply -f ${KUADRANT_REPO_RAW}/config/dependencies/istio/sail/istio.yaml
-else
-  # Create CRD first to prevent race condition with creating CR
-  info "Generating Istio configuration... 🛠️"
-  kubectl kustomize ${MGC_ISTIO_KUSTOMIZATION} >${TMP_DIR}/doctmp
```
Comment (david-martin) [inline hack/quickstart-setup.sh:461]: with this removed, you can remove `MGC_ISTIO_KUSTOMIZATION` & `MGC_REF` vars as well

**Comment by david-martin** [comment]
/lgtm
/approve

verified locally

**Comment by maleck13** [comment]
I think before we do this we need to figure out how we keep sail upto date. Really we should be testing against the latest sail upstream at the moment we are pinned to an older version for stability

**Comment by jasonmadigan** [comment]
@maleck13 Adam working a bit on this in https://github.com/Kuadrant/kuadrant-operator/pull/696/files

**Comment by maleck13** [comment]
This is already the case now

**Pull request closed without merging**